### DISCUSSION TITLE
Less PP obtained without FC

### DISCUSSION THREAD

**Discussion by nana123126** [ppy/osu#26762] [Ideas] [answered]
The PP system is not very friendly to players without FC, I think maybe some changes can be made? At least, come to Lazer and show players the brand new advanced PP system :)

How does the PP system look now?
--
When breaking a combo (especially in the middle of a map), 'aim pp' will significantly decrease. Breaking a combo in the middle of a map can cause 50% of aim pp loss, usually accompanied by a few misses, resulting in an extremely low aim pp. The only way to get a decent 'aim pp' is through 'Full Combo'. (Another drawback is that if the map only has difficulty at the beginning or end, the aim pp will not decrease significantly, but players may not necessarily have such high abilities.)

But without FC, does it necessarily mean aim is bad? In my opinion, if it weren't for FC and a map, I wouldn't have been able to get many pp. Even if my acc is good, a miss can significantly reduce aim pp, which is enough to undermine my confidence and then Ctrl+R.

I don't know how the uploaded score is deducted, but for local PP, I have observed that the only parameters that affect the deduction are the data in the settlement interface (meaning 100% acc=100% pp). If only these parameters are available, I believe that PP is only accurate when the player obtains SS, and other times it drops significantly due to conservatism. I don't really believe in PP without FC, which are too low. If the player has FC, So it is highly unlikely that it will be included in 'best performance'.

( Lazer's current score seems less affected by combo, I like it ) ( Perhaps it's because there's no need to compare scores with other maps? )

What are some ways to improve it?
--
So, what I mean is to create a higher confidence PP system and mix it with the original PP system. What I want is to make the score for entering 'best performance' a higher PP instead of a higher score, and then calculate the two PP systems (discard the lowest) on the local client after the player plays, check if they enter the top 100 of 'best performance', and then upload the score, For players who have not entered the top 100, using the PP provided by the player client can significantly reduce the score calculation for all players, as players cannot always improve.

For a 'higher confidence PP system', I think we only need to modify the original PP reduction method (the PP system cannot see what players have done while playing, but SS's PP is still very accurate. This way, there is no need to make drastic changes):

Remove the mechanism of using panel data to deduct PP.
For missing notes, it is important to analyze why they are missing. For example, if a large number of notes are too fast, there is no need to deduct aim pp (if the player keeps up).

Aim - For miss notes, check the distance between the click position and the note position to determine the deduction point. It is best to refer to the position of the previous note.
Speed - For miss notes, if the targeting difficulty is higher than before, it should be considered that it may be due to aim, similar to not scoring or deducting points, which is deducted to the previous speed pp (not increasing due to the maximum pp increase, as it has already been deducted).
Acc - For miss notes, check the number of hits to reduce uncertain pp caused by insufficient samples. Therefore, for acc pp that hit 20 or more, not much will be deducted.

Aim - For OK notes, check the distance between the next note and the longer distance, wait until the next note is hit, and then check again for the time consumed
Speed - For OK notes, clicking too quickly does not deduct points. (Considering double pressing, the maximum speed pp should be reduced)
Acc - For OK notes, first confirm if the speed has caused a decrease in acc. If the speed difficulty is higher than before, it is similar to not scoring or deducting points, which means deducting the previous acc pp.

As shown in the example above, it should be good.

SS's score is still considered 100% pp

To ensure real-time calculation, they all analyze the current note and existing PP to add deduction items. For the past, they have already been reflected in the deduction items. What I want is to accumulate deduction points during the gameplay process, and directly subtract them from the maximum pp calculated by the original pp system. However, it significantly slows down the calculation of the entire score, as it checks the player's behavior rather than panel parameters.

If we cannot achieve these, then we can only prove our PP through FC or SS.

**Comment by cihe13375** [comment]
Combo scaling removal (making pp not directly dependent on combo) has been purposed for a long time (https://github.com/ppy/osu/pull/16280), and is not merged yet mostly due to technical reasons (["dependent on pp highscores not being overwritten by higher scoring values"](https://github.com/orgs/ppy/projects/12/views/1); i.e. to avoid being scorev1'd)


> Remove the mechanism of using panel data to deduct PP.

In this way it would be impossible to recalculate pp on many old plays (whose replay files are not available, at least from the website)

**Comment by Rekunan** [comment, marked as answer]
I would also suggest joining the [pp dev discord](https://discord.gg/aqPCnXu) to have better visibility on your feedback.
